packages/mcp/src/tools/process-manager.ts

```diff
@@ -0,0 +1,596 @@
+  const proc = task.process;
+  proc.kill(signal ?? "SIGTERM");
+  const exitPromise = once(proc, "exit");
+
+  const guard = new Promise<never>((_resolve, reject) => {
+    const terminateTimer = setTimeout(() => {
+      if (!proc.killed) {
+        proc.kill("SIGKILL");
+      }
+      const forceTimer = setTimeout(() => {
+        reject(
+          new Error(
+            `Process ${proc.pid ?? "<unknown>"} failed to exit after SIGKILL`,
+          ),
+        );
+      }, TERMINATE_FORCE_TIMEOUT_MS);
+      forceTimer.unref();
+      void exitPromise.finally(() => {
+        clearTimeout(forceTimer);
+      });
+    }, TERMINATE_GRACE_PERIOD_MS);
+    terminateTimer.unref();
+    void exitPromise.finally(() => {
+      clearTimeout(terminateTimer);
+    });
+  });
+
+  await Promise.race([exitPromise, guard]);
+  await exitPromise;
+
+  const combined = `${task.stdoutText}${task.stderrText}`;
+  const tailText = combined.slice(-tail);
+  return { tail: tailText } as const;
+};
+
+const EnqueueSchema = z.object({
+  command: z.string(),
+  args: z.array(z.string()).default([]),
+  opts: z
+    .object({
+      name: z.string().optional(),
+      cwd: z.string().optional(),
+      env: z.record(z.string()).optional(),
+      timeoutDuration: z.number().positive().optional(),
+    })
+    .optional(),
+  name: z.string().optional(),
+  cwd: z.string().optional(),
+  env: z.record(z.string()).optional(),
+  timeout: z.boolean().optional(),
+  timeoutDuration: z.number().positive().optional(),
+});
+
+export const processGetTaskRunnerConfig: ToolFactory = () => ({
+  spec: {
+    name: "process.getTaskRunnerConfig",
+    description: "Return the current task runner configuration.",
+  },
+  invoke: async () => ({ config: getConfig() }),
+});
+
+export const processUpdateTaskRunnerConfig: ToolFactory = () => {
+  const Schema = z.object({
+    key: z.enum(["path", "maxRunning", "timeout"]),
+    value: z.union([z.string(), z.number()]).optional(),
+  });
+  return {
+    spec: {
+      name: "process.updateTaskRunnerConfig",
+      description: "Update a single key in the task runner configuration.",
+      inputSchema: {
+        key: Schema.shape.key,
+        value: Schema.shape.value,
+      },
+    },
+    invoke: async (raw: unknown) => {
+      const { key, value } = Schema.parse(raw);
+      const cfg = updateConfig(key, value);
+      return { config: cfg };
+    },
+  };
+};
+
+export const processEnqueueTask: ToolFactory = () => ({
+  spec: {
+    name: "process.enqueueTask",
+    description:
+      "Enqueue a command for execution respecting concurrency limits.",
+    inputSchema: {
+      command: z.string(),
+      args: z.array(z.string()).optional(),
+      opts: z
+        .object({
+          name: z.string().optional(),
+          cwd: z.string().optional(),
+          env: z.record(z.string()).optional(),
+          timeoutDuration: z.number().optional(),
+        })
+        .optional(),
+      name: z.string().optional(),
```

Comment:
Return a canonical handle here: `{ id: task.id, name: task.name, pid: task.pid }`. PIDs can be null/reused and names aren’t unique; callers should key off `id`.